service/create/domains.go

```diff
@@ -0,0 +1,111 @@
+package create
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/service/route53"
+	"github.com/giantswarm/aws-operator/resources"
+	awsresources "github.com/giantswarm/aws-operator/resources/aws"
+	"github.com/giantswarm/awstpr"
+	microerror "github.com/giantswarm/microkit/error"
+	"github.com/juju/errgo"
+)
+
+type hostedZoneInput struct {
+	Cluster awstpr.CustomObject
+	Client  *route53.Route53
+}
+
+type recordSetInput struct {
+	Cluster      awstpr.CustomObject
+	Client       *route53.Route53
+	Resource     resources.DNSNamedResource
+	Domain       string
+	HostedZoneID string
+}
+
+func (s *Service) createHostedZone(input hostedZoneInput) (*awsresources.HostedZone, error) {
+	hzName, err := hostedZoneName(input.Cluster)
+	if err != nil {
+		return nil, microerror.MaskAny(err)
+	}
+
+	hz := &awsresources.HostedZone{
+		Name:    hzName,
+		Comment: hostedZoneComment(input.Cluster),
+		Client:  input.Client,
+	}
+
+	hzCreated, err := hz.CreateIfNotExists()
+	if err != nil {
+		return nil, microerror.MaskAny(fmt.Errorf("error creating hosted zone '%s'", errgo.Details(err)))
```

Comment:
That looks weird. I think just `microerror.MaskAny(err)` does the trick. When you want to attach your additional custom message `microerror.MaskAnyf(err, "your additional custom message")` is what you want.